### PR TITLE
Controller/PPT to image endpoint

### PR DESCRIPTION
**Pull request by alszla**

### CODE REVIEW

File: src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @author Josue Rodriguez Garcia
+ * @version 1.0
+ */
+
+public class PPTtoPdfResponse {
+    private String fileName;
+    private String fileDownloadUri;
+    private String checksum;
+
```
Comment (LimbertVargas) [inline src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java]: maybe is necessary adding documentation for this method?

File: src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    private String fileName;
+    private String fileDownloadUri;
+    private String checksum;
+    
+    public PPTtoPdfResponse(String fileName, String fileDownloadUri, String checksum) {
+        this.fileName = fileName;
+        this.fileDownloadUri = fileDownloadUri;
+        this.checksum = checksum;
+    }
+
```
Comment (LimbertVargas) [inline src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java:36]: please adding documentation

File: src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    public String getFileName() {
+        return fileName;
+    }
+
```
Comment (LimbertVargas) [inline src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java:45]: please adding documentation

File: src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    public String getFileDownloadUri() {
+        return fileDownloadUri;
+    }
+
```
Comment (LimbertVargas) [inline src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java:52]: please adding documentation

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 5]
```diff
@@ -335,4 +335,146 @@ public KeyFrameResponse convertKeyFrame(@RequestParam("asset") MultipartFile ass
             return null;
         }
     }
+
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:701]: Please comment this method.

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 6]
```diff
@@ -335,4 +335,146 @@ public KeyFrameResponse convertKeyFrame(@RequestParam("asset") MultipartFile ass
...
+                + ".zip").toUriString();
+        String inputChecksumString = "";
+        InputStream inputProperties;
+        try {
+            inputProperties = new FileInputStream("application.properties");
+            properties.load(inputProperties);
+        } catch (IOException io) {
+            io.printStackTrace();
+        }
+
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:728]: Remove this line.

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 7]
```diff
@@ -335,4 +335,146 @@ public KeyFrameResponse convertKeyFrame(@RequestParam("asset") MultipartFile ass
...
+                        outputJson.getString("ext"));
+                criterion.setDestPath(properties.getProperty("file.downloadDir") + outputJson.getString("name") + "/");
+                criterion.setName(outputJson.getString("name"));
+                criterion.setExt("bmp");
+                criterion.setDpi(150);
+                criterion.setFormatColor("RGB");
+                PdfThumbnail pdf2thumb = new PdfThumbnail();
+                pdf2thumb.convert(criterion);
+            }
+
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:780]: Remove this line.

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 8]
```diff
@@ -335,4 +335,146 @@ public KeyFrameResponse convertKeyFrame(@RequestParam("asset") MultipartFile ass
...
+            FolderZipped.zipFolder(properties.getProperty("file.downloadDir") + outputJson.getString("name"));
+
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:782]: Remove this line.

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 9]
```diff
@@ -335,4 +335,146 @@ public KeyFrameResponse convertKeyFrame(@RequestParam("asset") MultipartFile ass
...
+                + ".zip").toUriString();
+        String inputChecksumString = "";
+        InputStream inputProperties;
+        try {
+            inputProperties = new FileInputStream("application.properties");
+            properties.load(inputProperties);
+        } catch (IOException io) {
+            io.printStackTrace();
+        }
+
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:806]: Remove this line.

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 10]
```diff
@@ -335,4 +335,146 @@ public KeyFrameResponse convertKeyFrame(@RequestParam("asset") MultipartFile ass
...
+                metaDataFile.writeJsonFile(convertedFile);
+            } else {
+                //Creation XMP
+                File convertedFile = new File(properties.getProperty("file.downloadDir") +
+                        outputJson.getString("name") + "/" + outputJson.getString("name") +
+                        outputJson.getString("ext"));
+                Metadata metaDataFile = new Metadata();
+                metaDataFile.writeXmpFile(convertedFile);
+            }
+
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:844]: Remove this line.

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 11]
```diff
@@ -335,4 +335,146 @@ public KeyFrameResponse convertKeyFrame(@RequestParam("asset") MultipartFile ass
...
+            FolderZipped.zipFolder(properties.getProperty("file.downloadDir") + outputJson.getString("name"));
+
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:846]: Remove this line.

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 12]
```diff
@@ -335,4 +335,146 @@ public KeyFrameResponse convertKeyFrame(@RequestParam("asset") MultipartFile ass
...
         }
     }
+
+    public PPTtoPdfResponse PPTtoPdf(@RequestParam("asset") MultipartFile asset, @RequestParam("input") String input,
+                                     @RequestParam("config") String config, @RequestParam("output") String output) {
+        JSONObject inputJson = new JSONObject(input);
+        JSONObject configJson = new JSONObject(config);
+        JSONObject outputJson = new JSONObject(output);
+        String fileName = uploadService.storeFile(asset);
+        String fileDownloadUri = ServletUriComponentsBuilder.fromCurrentContextPath().path("/download/").path(outputJson.getString("name")
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java]: Must have 120 line.

File: src/main/java/com/fundation/webservice/controller/Controller.java [hunk 13]
```diff
@@ -335,4 +335,146 @@ public KeyFrameResponse convertKeyFrame(@RequestParam("asset") MultipartFile ass
...
+        }
+    }
+
+    public PPTtoPdfResponse Image(@RequestParam("asset") MultipartFile asset, @RequestParam("input") String input,
+                                     @RequestParam("config") String config, @RequestParam("output") String output) {
+        JSONObject inputJson = new JSONObject(input);
+        JSONObject configJson = new JSONObject(config);
+        JSONObject outputJson = new JSONObject(output);
+        String fileName = uploadService.storeFile(asset);
+        String fileDownloadUri = ServletUriComponentsBuilder.fromCurrentContextPath().path("/download/").path(outputJson.getString("name")
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/controller/Controller.java:796]: Correct this line you must have 120 lines.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 6 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:728]: Adopted as this group standard, this line is inserted in order to make readibility easier. At this point we want to stress the difference between the operations carried before the line and after the line.
Conclusion: This blank line will be preserved.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 7 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:780]: Adopted as this group standard, this line is inserted in order to make readibility easier. At this point we want to stress the difference between the operations carried before the line and after the line.
Conclusion: This blank line will be preserved.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 8 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:782]: Adopted as this group standard, this line is inserted in order to make readibility easier. At this point we want to stress the difference between the operations carried before the line and after the line.
Conclusion: This blank line will be preserved.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 9 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:806]: Adopted as this group standard, this line is inserted in order to make readibility easier. At this point we want to stress the difference between the operations carried before the line and after the line.
Conclusion: This blank line will be preserved.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 11 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:846]: Adopted as this group standard, this line is inserted in order to make readibility easier. At this point we want to stress the difference between the operations carried before the line and after the line.
Conclusion: This blank line will be preserved.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 12 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java]: Fixed

File: src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java (hunk 1 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java]: Fixed.

File: src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java (hunk 2 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java:36]: Fixed.

File: src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java (hunk 3 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java:45]: Fixed.

File: src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java (hunk 4 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/PPTtoPdfResponse.java:52]: Fixed.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 5 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:701]: Fixed

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 10 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:844]: Adopted as this group standard, this line is inserted in order to make readibility easier. At this point we want to stress the difference between the operations carried before the line and after the line.
Conclusion: This blank line will be preserved.

File: src/main/java/com/fundation/webservice/controller/Controller.java (hunk 13 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/controller/Controller.java:796]: Fixed.

**Pull request merged**